### PR TITLE
Telecommands to delete elements from the agenda 

### PR DESCRIPTION
**Pull request by christopherkinyua**
I created two telecommands to:
- delete all agendas
- delete an agenda using a timestamp sent arg 

Comments
- I was not sure where to put the telecommands and I kept them in `telecommands_definitions.c`. Let me know the best practice for this. I did not want to put it into the `telecommands_executor.c` file.
- I wanted to know if there is a way to test the telecommand. I only tested it by ensuring I was able to see and call the telecommand. I was not able to validate I got rid of the telecommand agenda. Is there a way to test if I deleted an agenda?

### CODE REVIEW

**Comment by DeflateAwning** [comment]
Please move telecommands to a different file (named something like `agenda_telecommands.c/h`). Each category of telecommand should have its own file.

To test, you can schedule a telecommand for execution in the near future (e.g., `CTS1+hello_world()@tssent=1723170620007@tsexec=1000!`), and then try cancelling it by its tssent number.

Implements #109.

**Review by DeflateAwning — CHANGES_REQUESTED** [review]
Looks super solid! Just want to move these into a separate file, then can be approved! Good work :)

File: firmware/Core/Src/telecommands/telecommand_executor.c
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/telecommand_executor.c:191]: Just noticed it, but can you please change the `tssent` arg to be a regular into instead of a pointer.

File: firmware/Core/Src/telecommands/agenda_telecommands.c
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/agenda_telecommands.c:46]: Make const.

File: firmware/Core/Src/telecommands/agenda_telecommands.c
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/agenda_telecommands.c:37]: Make const.

File: firmware/Core/Src/telecommands/agenda_telecommands.c
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/agenda_telecommands.c:52]: Please move return here, and then unnest the `if (result == 0)` part

**Comment by christopherkinyua** [comment]
I've implemented the changes you suggested but wanted your insight based on the testing.

In the photo below, I have highlighted:
- The scheduled telecommand
- Attempt 1 of trying to delete the telecommand agenda
- Attempt 2 of trying to delete the telecommand agenda

My test process was:
1) Schedule a telecommand for 5000ms from the current epoch time which was 1723180260
2) Used the `agenda_delete_by_tssent` telecommand to delete the agenda using the tssent: 1723180260
3) Used the `agenda_delete_by_tssent` telecommand to delete the agenda using the tssent: 1723180260 added with the 5000 offset which is: 1723185260

The result was that they both failed and from my understanding, the first one should passed and be deleted from the agenda. I was just curious to see what your thoughts are and if something is missing in my process.

agenda_redline

**Review by DeflateAwning — CHANGES_REQUESTED** [review]
Sorry I didn't catch these comments in the review earlier. Couple of tiny changes

**Comment by DeflateAwning** [comment]
Regarding your testing procedure, `@tssent=5000` isn't an offset; it's an absolute unix timestamp in ms. As you can see, the command is executed immediately because the system uptime (which matches the current unix timestamp, until the `set_unix_timestamp_ms` telecommand (or whatever it's called) is run.

File: firmware/Core/Src/telecommands/agenda_telecommands.c
Comment (christopherkinyua) [reply firmware/Core/Src/telecommands/agenda_telecommands.c:46]: Done

File: firmware/Core/Src/telecommands/agenda_telecommands.c
Comment (christopherkinyua) [reply firmware/Core/Src/telecommands/agenda_telecommands.c:37]: Done

File: firmware/Core/Src/telecommands/agenda_telecommands.c
Comment (christopherkinyua) [reply firmware/Core/Src/telecommands/agenda_telecommands.c:52]: Done

File: firmware/Core/Src/telecommands/telecommand_executor.c
Comment (christopherkinyua) [reply firmware/Core/Src/telecommands/telecommand_executor.c:191]: Done, I removed the pointer tag from the function and the function definition and passed the `const uint8_t tssent` variable into the function instead of a pointer.

Is there a reason I should avoid using a pointer for that instant?

**Comment by christopherkinyua** [comment]
I'm trying to make sense of this. There are two tags that were passed: `tssent` which is the time the telecommand was sent and `tsexec` which I think is the time the telecommand should execute after time has passed.

So during the testing, I was doing, I was passing the current epoch time into `tssent` and adding the 5000 to the `tsexec`. The way I made the function is that it would use the `tssent` that I passed to be checked and then delete the agenda.

Unless the epoch time on the dev board is different, is there something I'm not getting? Like in my test process or the values I'm passing in?

File: firmware/Core/Src/telecommands/telecommand_executor.c
Comment (DeflateAwning) [reply firmware/Core/Src/telecommands/telecommand_executor.c]: Using a pointer is unnecessary, allows for the introduction of accidental errors. Additionally, it prevents the value from being inadvertently modified from within the function.

**Comment by DeflateAwning** [comment]
The `tssent` argument is currently mostly just used as a unique ID of each telecommand sent, currently. 

The board assumes it's 1970-01-01T00:00:00Z when it powers on. You may find it helpful to use the `CTS1+set_system_time(time_ms)!` command to set the time to the real current time for realistic testing. 

When the board powers on, its uptime is 0, and its unix epoch time is 0. Thus, the uptime (echoed in the heartbeat messages) and the unix timestamp are equal until the unix timestamp is set via the `set_system_time` telecommand, or by the GPS.

File: firmware/Core/Src/telecommands/telecommand_executor.c
Comment (christopherkinyua) [reply firmware/Core/Src/telecommands/telecommand_executor.c]: Thanks that makes sense

**Comment by christopherkinyua** [comment]
Ahh I sort of get it. So the `tssent` is like an ID as you said and the `tsexec` is the amount of time in ms that should pass then to run the telecommand which is relative to the system uptime.

Thanks, I was able to run a successful test of the telecommand as seen below. Im gonna wait for your approval then I can squash and merge.

agenda

**Comment by DeflateAwning** [comment]
> tsexec is the amount of time in ms that should pass then to run the telecommand

No. `tsexec` is an absolute unix timestamp (number of miliseconds since 1970-01-01T00:00:00) at which to execute.

**Comment by christopherkinyua** [comment]
Ahh yes, that's right. I failed to explain it like that.

**Comment by DeflateAwning** [comment]
To facilitate testing and review, I'm making a couple changes/additions:

* `agenda_telecommands.c` and `agenda_telecommand_defs.h` should have the same name stem (probably `agenda_telecommand_defs` for consistency). I'm renaming `agenda_telecommands.c` to `agenda_telecommand_defs.c`.
* `TCMD_agenda_delete_all` cannot fail, and its failure code is not checked, so it is changed to a void function. 
* Added a log message showing which command and tssent/tsexec value is being executed from the agenda. 
* Improved logging within the two functions you created (now shows the name of the telecommand which is deleted from the agenda). 

Please review these changes and let me know if you have any questions or comments!

Thank you again for working on this!

**Review by DeflateAwning — APPROVED** [review]
After you've checked through the changes I tacked on, please squash-and-merge!

**Comment by christopherkinyua** [comment]
It looks good and it makes sense to me. I see you used the `LOG_message()` function. I'll start using it now too. Ill squash and merge.

**Comment by DeflateAwning** [comment]
Thanks a lot!

There'll be announcement about the new `LOG_message()` function soon! Just wanted to start using it a bit, as it shows the timestamp (which was useful to debug when the commands were/weren't being executed here).

**Pull request merged**
